src/lexer.c

```diff
@@ -1,109 +1,130 @@
 /* ************************************************************************** */
 /*                                                                            */
 /*                                                        :::      ::::::::   */
-/*   lexer.c                                            :+:      :+:    :+:   */
+/*   lexer_list.c                                       :+:      :+:    :+:   */
 /*                                                    +:+ +:+         +:+     */
 /*   By: sqiu <sqiu@student.42vienna.com>           +#+  +:+       +#+        */
 /*                                                +#+#+#+#+#+   +#+           */
-/*   Created: 2023/06/23 13:13:28 by sqiu              #+#    #+#             */
-/*   Updated: 2023/07/21 11:51:37 by sqiu             ###   ########.fr       */
+/*   Created: 2023/06/26 16:57:22 by sqiu              #+#    #+#             */
+/*   Updated: 2023/07/21 15:57:04 by sqiu             ###   ########.fr       */
 /*                                                                            */
 /* ************************************************************************** */
 
-#include "lexer.h"
+/**
+ * @file lexer_list.c
+ * @brief Contains functions to create a double-linked list of tokens.
+ */
+
+#include "mod_lexer.h"
+/* #include "lexer_list.h"
+#include "lexer_utils.h"
+#include "lexer_tok_utils.h" */
 
 /**
- * @brief Iterates through input string and categorises character sequences
- * into tokens.
- * 
- * Spaces are being skipped.
- * The input is divided into <simple commands> which are separated by
- * pipes '|' and may consist of the strings
- * 
- * 		<infile exe opt1 opt2 opt3 >outfile
- * 
- * Each of these input strings represent a token. These tokens are saved
- * according to each simple command. 
- * 
- * @param data 	Overarching struct to hold all necessarry data.
- * @param input Input string delivered by <readline>.
- * @return t_err 
+ * @brief Create a list of token.
```

Comment:
That's basically it :D what else should there be

Reply:
I think the while loop (err != EOF || !*lst_head) could be explained. Why are we looping if we have no head? Wouldn't this also loop if ft_tokenize() failed?

Reply:
It loops even w/o head so that a node gets created even if the first token is directly at the end of file (no newline). err would be ERR_EOF. Without the head condition that token would not be saved in the list. If ft_tokenise() fails, it would loop but break out with an error with ft_new_node()

Reply:
I'd suggest add a shorter version of this to the doxy? Like:
Create token with ft_tokenize(). 
At least one node gets created, since lst_head is empty on first run. 
Create tokens in loop until EOF

Reply:
Nice comment!

Reply:
Comment elaborated